src/map/script.c

```diff
@@ -9228,12 +9167,33 @@ static BUILDIN(getitemgroupitems)
 		st->state = END;
 		return false;
 	}
-	
+
+	int nameid = script_getnum(st, 3);
+	struct item_data *item = itemdb->exists(nameid);
+	if (item == NULL) {
+		ShowError("buildin_getitemgroupitems: invalid item %d\n", nameid);
+		script_pushint(st, 0);
+		return false;
+	}
+
+	const struct item_group *group = itemdb->search_group(item->nameid);
+	if (group == NULL) {
+		ShowWarning("buildin_getitemgroupitems: item group not found\n");
+		script_pushint(st, 0);
+		return false;
+	}
+
 	int count = 0;
-	for (int i = 0; i < group.qty; i++) {
-		int id = group.nameid[i];
+	for (int i = 0; i < group->qty; i++) {
+		int id = group->nameid[i];
+		int j = 0;
+		ARR_FIND(0, i, j, group->nameid[i] == group->nameid[j]);
+		if (i != j) {
+			// Already encountered - skip duplicates
```

Comment:
so it's possible to have duplicates in item_group stuff but this script command will ignore that fact? Is there no way to combine duplicate entries into one for this? Or do duplicates really have no effect?

Reply:
Duplicates affect the probability that a given item is selected. But we're returning a bare list of IDs, without caring about the probability. We could return both, but then we'd need the user to either pass two arrays, or to return an array of interleaved IDs and counts. I kept the original behavior from the PR author of only returning the IDs for now

Reply:
Ah, okay. As long as the list doesn't contain information about the chances, it's fine.